bundles/net.tourbook/src/net/tourbook/extension/export/ActionExport.java

```diff
@@ -150,28 +150,30 @@ public void dispose() {
                   TourbookPlugin.PLUGIN_ID,
                   TourbookPlugin.EXT_POINT_EXPORT_TOUR);
 
-      if (extPoint != null) {
+      if (extPoint == null) {
+         return _exportExtensionPoints;
+      }
 
-         for (final IExtension extension : extPoint.getExtensions()) {
-            for (final IConfigurationElement configElement : extension.getConfigurationElements()) {
+      for (final IExtension extension : extPoint.getExtensions()) {
+         for (final IConfigurationElement configElement : extension.getConfigurationElements()) {
 
-               if (configElement.getName().equalsIgnoreCase("export")) { //$NON-NLS-1$
-                  try {
-                     final Object object = configElement.createExecutableExtension("class"); //$NON-NLS-1$
-                     if (object instanceof ExportTourExtension) {
+            if (!configElement.getName().equalsIgnoreCase("export")) { //$NON-NLS-1$
+               continue;
+            }
```

Comment:
Does it make it more readable if I do this ?

![image](https://user-images.githubusercontent.com/25235665/107134540-37861900-68b0-11eb-96f1-f1ead68d56ed.png)

Reply:
When there is a simple logical structure, like in this case, there is no need to "optimize" it.

* The first if is changed from `!= null` into `== null`
* The second if is changed from a none negation into a negation

![mt-simple-logical-structure](https://user-images.githubusercontent.com/1283445/107140056-e7469f80-691f-11eb-947a-7895add3f896.png)

Reply:
I can put it back if you really want to but my vision was that it's bloated code (if -> for-> for -> if......) and it makes it look to the right and is not readable.
The code on the left is more readable because it is not so much indented to the right
let me know

![image](https://user-images.githubusercontent.com/25235665/107170511-bf862480-697d-11eb-9d05-5d8fa44a49a2.png)